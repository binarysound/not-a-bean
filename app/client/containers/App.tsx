import React from 'react'
import io from 'socket.io-client'

namespace App {
  /* tslint:disable-next-line:no-empty-interface */
  export interface IProps {}

  export interface IState {
    message: string
  }
}

export class App extends React.Component<App.IProps, App.IState> {
  constructor(props: App.IProps) {
    super(props)

    this.state = {
      message: 'No message received.',
    }
  }

  public componentDidMount() {
    const socket = io.connect('/')
    socket.on('server event', (data: any) => {
      this.setState({
        message: data.message,
      })
      socket.emit('client event', { my: 'data' })
    })
  }

  public render() {
    return (
      <div>{this.state.message}</div>
    )
  }
}
