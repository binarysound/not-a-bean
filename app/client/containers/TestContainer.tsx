import { push } from 'connected-react-router'
import React from 'react'
import { connect, DispatchProp } from 'react-redux'
import { Route, Switch } from 'react-router'

import { AppAction } from '@/client/actions'
import { Screen } from '@/client/components/Screen'
import { IAppState } from '@/models/appState'
import { NotABean } from '@/not-a-bean/client/components/NotABean'

const Page = (props: {
  dispatch: DispatchProp<AppAction>['dispatch'],
  pageName: string,
  toPageName: string,
  toPath: string,
  children?: React.ReactNode,
}) => (
  <div>
    <div>{props.pageName}</div>
    {props.children}
    <div
      style={{
        color: 'blue',
        cursor: 'pointer',
      }}
      onClick={() => props.dispatch(push(props.toPath))}
    >
      to {props.toPageName}
    </div>
  </div>
)

class _TestContainer extends React.Component<_TestContainer.IProps> {
  public render() {
    const { dispatch } = this.props
    return (
      <Screen>
        <Switch>
          <Route
            path='/'
            exact={true}
            render={() => (
              <Page
                dispatch={dispatch}
                pageName='Main page'
                toPageName='Another page'
                toPath='/another'
              >
                {null}
              </Page>
            )}
          />
          <Route
            path='/another'
            exact={true}
            render={() => (
              <Page
                dispatch={dispatch}
                pageName='Another page'
                toPageName='Main page'
                toPath='/'
              />
            )}
          />
          <Route
            path='/not-a-bean'
            component={NotABean}
          />
          <Route
            render={() => (
              <Page
                dispatch={dispatch}
                pageName='404'
                toPageName='Main page'
                toPath='/'
              />
            )}
          />
        </Switch>
      </Screen>
    )
  }
}

namespace _TestContainer {
  export interface IPropsFromState {
    currentPath: string  // prop to force rerender when path changes
  }
  /* tslint:disable-next-line:no-empty-interface */
  export interface IOwnProps {}

  export type IProps = IPropsFromState & IOwnProps & DispatchProp<AppAction>

  export function mapStateToProps(state: IAppState): IPropsFromState {
    return {
      currentPath: state.router.location.pathname,
    }
  }
}

export const TestContainer = connect<
  _TestContainer.IPropsFromState,
  DispatchProp<AppAction>,
  _TestContainer.IOwnProps,
  IAppState
>(_TestContainer.mapStateToProps)(_TestContainer)
