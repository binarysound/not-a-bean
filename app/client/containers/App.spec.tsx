import React from 'react'
import renderer from 'react-test-renderer'

import { App } from '@/client/containers/App'

describe('App', () => {
  it('renders correctly', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
