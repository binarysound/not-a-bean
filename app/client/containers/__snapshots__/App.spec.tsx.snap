// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  No message received.
  <div>
    <div>
      Main page
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "color": "blue",
          "cursor": "pointer",
        }
      }
    >
      to 
      Another page
    </div>
  </div>
</div>
`;
